optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -143,14 +148,44 @@ public void solve(DefaultSolverScope<Solution_> solverScope) {
                 stepEnded(stepScope);
                 phaseScope.setLastCompletedStepScope(stepScope);
             }
-        } finally {
+        } catch (RuntimeException e) {
             // If a partition thread throws an Exception, it is propagated here
             // but the other partition thread won't finish any time soon, so we need to terminate them
             childThreadPlumbingTermination.terminateChildren();
+            throw e;
+        } finally {
+            List<Runnable> partitionSolvers = threadPoolExecutor.shutdownNow();
+            logger.trace("{}         Awaiting termination of {} partition solvers.",
```

Comment:
invalid because it's only the non started partitioned solvers, so remove this log message